### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by elioschmutz**
@maethu 

First pullrequest with feature-implementation from ftw.contentpage

### CODE REVIEW

**Comment by elioschmutz** [comment]
@maethu updated the following:
- remove not working uninstall functionality
- remove browser-layer
- refactoring configure.zcml
- adjust readme for behavior
- change indexer getContentCategories to get_content_categories

File: ftw/contentnav/behaviors/content_categories.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+
+alsoProvides(IContentCategories, IFormFieldProvider)
+
+
+class ContentCategoriesStorage(object):
+
+    def __init__(self, context):
+        self.context = context
+        self.annoation_storage = AnnotationsFactoryImpl(
```
Comment (mbaechtold) [inline ftw/contentnav/behaviors/content_categories.py]: s/annoation_storage/annotation_storage

File: ftw/contentnav/configure.zcml [hunk 2]
```diff
@@ -2,9 +2,45 @@
...
+    <!-- include package dependencies -->
+    <include package="z3c.autoinclude" file="meta.zcml" />
+    <includeDependencies package="." />
+
+    <include package="ftw.profilehook" />
+
+    <i18n:registerTranslations directory="locales" />
+
+    <include file="permissions.zcml" />
+	<include package=".behaviors" />
```
Comment (mbaechtold) [inline ftw/contentnav/configure.zcml:19]: Could you please use the same indentation level? There are more cases in this file, not just this one.

File: ftw/contentnav/testing.py [hunk 3]
```diff
@@ -1,44 +1,64 @@
...
+class SampleBuilder(DexterityBuilder):
+    portal_type = 'Sample'
+
+registry.builder_registry.register('sample', SampleBuilder)
+
+
+FTW_CONTENTNAV_FIXTURE = FtwContentnavLayer()
 FTW_CONTENTNAV_INTEGRATION_TESTING = IntegrationTesting(
-    bases=(FTW_CONTENTNAV_FIXTURE,), name="FtwCONTENTNAV:Integration")
+    bases=(FTW_CONTENTNAV_FIXTURE,), name="FtwSimplelayout:Integration")
```
Comment (mbaechtold) [inline ftw/contentnav/testing.py]: Why name it "simplelayout"?

File: ftw/contentnav/viewlets/contentnavlisting.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            'object_provides': 'ftw.contentnav.interfaces.ICategorizable'}
+        contents = self.context.getFolderContents(contentFilter=query,
+                                                  full_objects=True)
+        return self._create_resultmap(contents)
+
+    def _create_resultmap(self, contents=None):
+        resultmap = {}
+        categories =[]
+        if not contents:
+            categories
```
Comment (mbaechtold) [inline ftw/contentnav/viewlets/contentnavlisting.py]: This statement seems to have no effect.

**Comment by elioschmutz** [comment]
@mbaechtold changes done!

please, could you take another look on it...?

File: ftw/contentnav/viewlets/contentnavlisting.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        query = {
+            'sort_on': 'sortable_title',
+            'object_provides': 'ftw.contentnav.interfaces.ICategorizable'}
+        contents = self.context.getFolderContents(contentFilter=query,
+                                                  full_objects=True)
+        return self._create_resultmap(contents)
+
+    def _create_resultmap(self, contents=None):
+        resultmap = {}
+        categories =[]
```
Comment (mbaechtold) [inline ftw/contentnav/viewlets/contentnavlisting.py]: Missing space (PEP 8)

**Comment by mbaechtold** [comment]
Currently the categories are also shown on the edit form. ~~As discussed, please remove them.~~ (we leave that out for now).

**Comment by mbaechtold** [comment]
Otherwise :+1: (it works on my machine).

**Comment by elioschmutz** [comment]
@mbaechtold 

Changes done!

**Pull request merged**